tests/bluetooth/bsim_bt/bsim_test_rpa_peripheral/src/dut.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_bt_utils.h"
+#include "zephyr/bluetooth/addr.h"
+#include "zephyr/bluetooth/bluetooth.h"
+#include "zephyr/bluetooth/conn.h"
+#include "zephyr/toolchain/gcc.h"
+#include <host/hci_core.h>
+
+#include <stdint.h>
+#include <string.h>
+
+static struct bt_le_ext_adv_start_param start_params;
+static struct bt_le_ext_adv_cb cb_adv;
+static struct bt_le_ext_adv *adv;
+
+bool cb_rpa_expired(struct bt_le_ext_adv *adv)
+{
+	backchannel_sync_send();
+
+	/* Return true to rotate the current RPA */
+	return true;
+}
+
+void start_advertising(void)
+{
+	int err;
+
+	/* Enable bluetooth */
+	err = bt_enable(NULL);
+	ASSERT(!err, "Failed to enable bluetooth (err %d\n)", err);
+
+	/* Start non-connectable extended advertising with private address */
+	start_params.timeout = 0;
+	start_params.num_events = 0;
+	cb_adv.rpa_expired = cb_rpa_expired;
+
+	err = bt_le_ext_adv_create(BT_LE_EXT_ADV_CONN_NAME, &cb_adv, &adv);
+	ASSERT(!err, "Failed to create advertising set (err %d)\n", err);
+
+	err = bt_le_ext_adv_start(adv, &start_params);
+	ASSERT(!err, "Failed to enable periodic advertising (err %d)\n", err);
```

Comment:
What's the rationale of using `ASSERT` instead of `FAIL`?

Reply:
Using `ASSERT` for readability.

Do you recommend using this syntax instead everywhere `ASSERT` is used ?
```
if (!(expr)) {   
   FAIL("error message");
}

Reply:
I'm not sure what I recommend, but `FAIL` is mostly what we use in the babblesim tests. In case that `CONFIG_ASSERT=n`, I guess the error values will never be checked? 

IMO it shouldn't be possible to (purposely or accidentally) remove the test checks via a Kconfig

Reply:
Good point. 
Changed to `FAIL`